src/datepicker/datepicker.component.ts

```diff
@@ -0,0 +1,133 @@
+import { Component, Input, Output, EventEmitter } from "@angular/core";
+import { FlatpickrOptions } from "ng2-flatpickr";
+import rangePlugin from "flatpickr/dist/plugins/rangePlugin";
+
+let nextId = 0;
```

Comment:
This should be a static property on the class `private static datePickerCount = 0;` for example